### PR TITLE
Support hidden and virtual columns

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/activerecord-cockroachdb-adapter/issues/203

This allows us to fix issue with implicit rowid column being created
incorrectly when reloading a schema.

### CODE REVIEW

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb [hunk 1]
```diff
@@ -500,8 +503,18 @@ def column_definitions(table_name)
             dtype = field[1]
             field[1] = crdb_fields[field[0]][2].downcase if re.match(dtype)
             field[7] = crdb_fields[field[0]][1]&.gsub!(/^\'|\'?$/, '')
+            field[9] = true if crdb_fields[field[0]][3]
```
Comment (otan) [inline lib/active_record/connection_adapters/cockroachdb_adapter.rb:506]: this is so hacky, lol!

**Review by otan — APPROVED** [review]

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb (hunk 1 above)
Comment (rafiss) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb:506]: yeah idk, kinda surprising that there's no way to access the fields by name!

Comment (otan) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb:506]: just the whole `NULL as is_hidden` then overwriting it later is *chefs kiss*

**Comment by sandstrom** [comment]
Awesome @rafiss 🎉 🙏🏻

**Pull request merged**